samcli/commands/local/lib/local_lambda.py

```diff
@@ -205,6 +207,22 @@ def _make_env_vars(self, function):
             aws_creds=aws_creds,
         )
 
+    def _get_session_creds(self):
```

Comment:
we could make this a lazy property in the future.

Reply:
Good idea. Going to leave it as is for now but something we should consider for the future when dealing with things like this.